app/components/sidebars/main/channels_list/list/list.js

```diff
@@ -148,12 +154,18 @@ export default class List extends PureComponent {
                 id: t('sidebar.types.recent'),
                 defaultMessage: 'RECENT ACTIVITY',
             };
+        case 'channels':
         case SidebarSectionTypes.ALPHA:
             return {
                 action: this.showCreateChannelOptions,
                 id: t('mobile.channel_list.channels'),
                 defaultMessage: 'CHANNELS',
             };
+        case 'custom':
+            return {
+                id: displayName,
+                defaultMessage: displayName,
+            };
```

Comment:
Rather than have these as part of the existing list; I have in my stash a `buildCategorySections` - essentially also a smaller switch statement with a few tweaks just for displaying the category sections - wondering if it's worth having it out separately or okay to have it in the same list.

Reply:
this is basically used for localization, which I doubt is used with the categories, perhaps you should have a different function that gets called to build the sections in a different way than the legacy sidebar... in fact I would feel more comfortable if we mount a component for the legacy sidebar and a new one for the sidebar that has the categories, unless you think this is best. One of the reasons is because the sections are rendered differently